### DISCUSSION TITLE
git action button for "push" instead of "sync"

### DISCUSSION THREAD

**Discussion by nik-sm** [microsoft/vscode-discussions#501] [Extension Development QnA] [answered]
I like using the "commit" button in the Source Control pane, but I would prefer to swap the "sync" button for a simple "push" button. 

My reasoning is - if there are upstream changes, I would rather get an error message and have the chance to browse, before deciding if I'm ready to merge and push.
Thus I always either push from command line or use the 3-dots "more" dropdown and navigate to the small "push" item that is listed.

Is it possible to configure this behavior?

I found some related settings in preferences by searching "git action button" (under the setting id `git.showActionButton`), but it looks like what I'm describing isn't currently offered out-of-the-box.

Is this possible to customize myself, and/or is it possible to request this as a feature?

**Comment by gjsjohnmurray** [comment, marked as answer]
This doesn't belong here in the "Extension Development QnA" category of vscode-discussions.

I suggest you take it to the main repo. Here is a query that finds related issues there:

https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Agit+%22action+button%22

**Reply by nik-sm** [reply]
Ok I'll try over there. Thanks.
